gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/metrics/KafkaReportingFormats.java

```diff
@@ -104,6 +113,10 @@ public ScheduledReporter buildEventsReporter(String brokers, String topic, Metri
       if (Boolean.valueOf(properties.getProperty(ConfigurationKeys.METRICS_REPORTING_KAFKA_USE_SCHEMA_REGISTRY,
           ConfigurationKeys.DEFAULT_METRICS_REPORTING_KAFKA_USE_SCHEMA_REGISTRY))) {
         builder.withSchemaRegistry(new KafkaAvroSchemaRegistry(properties));
+        String schemaId = properties.getProperty(ConfigurationKeys.METRICS_REPORTING_EVENTS_KAFKA_AVRO_SCHEMA_ID);
```

Comment:
Repeated patterns: I am fine with it if you think current way is more readable

Reply:
I did try to avoid duplication here. The problem stems from the fact that the builders are of different classes and the common parent does not expose the methods that would have avoided the duplication.